app/Http/Controllers/ItemController.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Models\Item;
+use App\Interfaces\ImageStorage;
+use Illuminate\Http\RedirectResponse;
+use Illuminate\Http\Request;
+use Illuminate\View\View;
+
+class ItemController extends Controller
+{
+    public function index(): View
+    {
+        $viewData = [];
+        $viewData['items'] = Item::all();
+
+        return view('admin.item.index')->with('viewData', $viewData);
+    }
+
+    public function show(string $id): View|RedirectResponse
+    {
+        $viewData = [];
+        $item = Item::find($id);
+        if (is_null($item)) {
```

Comment:
Try to implement the try-catch alternative
```suggestion
       try {
            $viewData = [];
            $item = Item::findOrFail($id);
            $viewData['item'] = $item;

            return view('admin.item.show')->with('viewData', $viewData);
        } catch (Exception $e) {
            return redirect()->route('admin.index');
        }
```